crates/sqlexec/src/extension_codec.rs

```diff
@@ -978,6 +1061,73 @@ impl<'a> PhysicalExtensionCodec for GlareDBExtensionCodec<'a> {
                 show_statistics: true,
                 schema: Some(exec.schema().try_into()?),
             })
+        } else if let Some(exec) = node.as_any().downcast_ref::<WindowAggExec>() {
+            let input_schema: datafusion_proto::protobuf::Schema =
+                exec.input_schema().as_ref().try_into()?;
+
+            let window_expr = exec
+                .window_expr()
+                .iter()
+                .map(|e| try_from_window_expr(e.clone()))
+                .collect::<Result<Vec<protogen::sqlexec::physical_plan::GlarePhysicalWindowExprNode>>>()?;
+
+            let partition_keys = exec
+                .partition_keys
+                .iter()
+                .map(|e| e.clone().try_into())
+                .collect::<Result<Vec<datafusion_proto::protobuf::PhysicalExprNode>>>()?;
+            proto::ExecutionPlanExtensionType::WindowFunc(
+                protogen::sqlexec::physical_plan::WindowAggExecNode {
+                    // input: Some(Box::new(input)),
+                    window_expr,
+                    input_schema: Some(input_schema),
+                    partition_keys,
+                    partition_search_mode: None,
+                },
+            )
+        } else if let Some(exec) = node.as_any().downcast_ref::<BoundedWindowAggExec>() {
+            println!("{exec:#?}");
```

Comment:
should we be printing from this code?

Reply:
Since I cannot completely implement this here, (due to lack of stuff from datafusion), added this to modify the code accordingly.

This will be removed in the final implementation, but I doubt it would be this PR to do it.